pkg/proxy/healthcheck/service_health.go

```diff
@@ -226,24 +235,26 @@ func (h hcHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 	count := svc.endpoints
+	kubeProxyHealthy := h.hcs.healthzServer.IsHealthy()
 	h.hcs.lock.RUnlock()
 
 	resp.Header().Set("Content-Type", "application/json")
 	resp.Header().Set("X-Content-Type-Options", "nosniff")
-	if count == 0 {
-		resp.WriteHeader(http.StatusServiceUnavailable)
-	} else {
+	if count != 0 && kubeProxyHealthy {
 		resp.WriteHeader(http.StatusOK)
+	} else {
+		resp.WriteHeader(http.StatusServiceUnavailable)
 	}
 	fmt.Fprint(resp, strings.Trim(dedent.Dedent(fmt.Sprintf(`
 		{
 			"service": {
 				"namespace": %q,
 				"name": %q
 			},
-			"localEndpoints": %d
+			"localEndpoints": %d,
+			"kubeProxyHealthy": %v
```

Comment:
If we expect this to be an API that different proxy implementations could implement, maybe we should call this `dataplaneHealthy` or something?  Specifically not "kubeProxy" ?

Reply:
At this point this isn't library code so I wouldn't consider it an API. That being said: I do know some proxy implementations copy this code, so...sure, I'll update